src/types/type_param.rs

```diff
@@ -123,11 +95,12 @@ pub enum TypeArgError {
     // For now this only happens at the top level (TypeArgs of op/type vs TypeParams of Op/TypeDef).
     // However in the future it may be applicable to e.g. contents of Tuples too.
     #[error("Wrong number of type arguments: {0} vs expected {1} declared type parameters")]
-    WrongNumber(usize, usize),
-    /// The type declared for a TypeParam was an alias that was not resolved to an actual type
-    #[error("TypeParam required an unidentified alias type {0}")]
-    NoAliases(String),
-    /// There was some problem fitting a const int into its declared size
-    #[error("Error with int constant")]
-    Int(#[from] ConstIntError),
+    WrongNumberArgs(usize, usize),
```

Comment:
Following from previous suggestion - I actually thought you could move this out of TypeArgError into SignatureError. But I'm also quite happy not to, it might be nicer to keep it here